squad-get-log-lines

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python3
+
+import argparse
+import logging
+import os
+import sys
+from pathlib import Path
+
+from squad_client.core.api import SquadApi
+from squad_client.core.models import ALL, Squad
+
+squad_host_url = "https://qa-reports.linaro.org/"
+SquadApi.configure(cache=3600, url=os.getenv("SQUAD_HOST", squad_host_url))
+
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="Script to scrape the log parameter from tests in specified projects."
+    )
+
+    parser.add_argument(
+        "--group",
+        required=True,
+        help="The name of the SQUAD group.",
+    )
+
+    parser.add_argument(
+        "--projects",
+        required=True,
+        action="extend",
+        nargs="+",
+        help="A list of SQUAD projects to fetch logs from.",
+    )
+
+    parser.add_argument(
+        "--suites",
+        required=True,
+        action="extend",
+        nargs="+",
+        help="A list of SQUAD test suite names to get logs for.",
+    )
+
+    parser.add_argument(
+        "--debug",
+        required=False,
+        action="store_true",
+        default=False,
+        help="Display debug messages.",
+    )
+
+    parser.add_argument(
+        "--count",
+        required=False,
+        default=2,
+        type=int,
+        help="The number of builds to fetch when fetching logs.",
+    )
+
+    return parser.parse_args()
+
+
+def run():
+    args = parse_args()
+
+    if args.debug:
+        logger.setLevel(level=logging.DEBUG)
+
+    group_name = args.group
+    project_names = args.projects
+
+    group = Squad().group(group_name)
+    for project_name in project_names:
+        project = group.project(project_name)
+        builds = project.builds(count=args.count)
+
+        for build_id, build in builds.items():
+            # There will only be a failure log when the test does not pass
+            filters = {"result": False}
+
+            # Look at all the tests in the build
+            tests = build.tests(count=ALL, **filters)
+
+            for test_id, test in tests.items():
+                # only look at the suites we have requested - this is the first
+                # part of the name before the slash for example
+                # "ltp-syscalls/linkat02" has the suite "ltp-syscalls"
+                if test.name.split("/")[0] in args.suites:
+                    logger.debug(test.log)
+                    if test.log:
+                        log_file = (
+                            Path("logs")
+                            / Path(test.name)
+                            / Path(project.slug)
+                            / Path(str(test.id))
```

Comment:
I've added a few more patches to this PR (one patch to add support for log file download and a couple to deal with authentication headers for the file downloads).

To summarise, I've:
- Added support for downloading the full logs
- When full logs are downloaded, a directory structure is created which creates symlinks to the full logs next to symlinks to all the log lines which came from that full log
- Tweaked the initial patch slightly so it aligns a bit better with the above changes

I thought this way it would be easier to see the relationship between the original log and all the different log lines that came from that log.

Let me know what you think :+1: